### ISSUE TITLE
Merge Server and ServerOptions

### ISSUE THREAD

**Issue by bernardnormier** [icerpc/icerpc-csharp#182]
I am proposing to merge the Server and ServerOptions class into a single class, Server.

From example:
```
// will listen on localhost on the default port
var server = new Server() { Dispatcher = new Greeter(), Endpoint = "ice+tcp://127.0.0.1" };
await server.ListenAndServeAsync();
```

i.e.:
- Server provides a parameterless constructor and various get/set properties for Endpoint (a string), ProxyOptions etc. (just like the current ServerOptions)
- Once you call `ListenAndServeAsync` on the server, you can no longer set any of these properties
- ListenAndServerAsync returns the ShutdownComplete task.

server also provides a method Serve:
```
void Serve(Connection connection);
```
which "gives" a connection to this server for servicing (and "attaches" the server to this connection). That's for bi-dir servicing.

The inspiration is the go http server: see
https://golang.org/pkg/net/http/
https://golang.org/pkg/net/http/#Server

In summary:

```
public class sealed Server : IAsyncDisposable
{
      // all ServerOptions properties, including Dispatcher
 
      public Task ShutdownComplete { get; }

      public Server() {} // parameterless

      public Task ListenAndServeAsync(CancellationToken cancel = default) { ... }
      public void Serve(Connection connection) { ... }
      public Task ShutdownAsync(CancellationToken cancel = default) { ... }
}
```
